puffin/src/scope_details.rs

```diff
@@ -0,0 +1,261 @@
+use std::{borrow::Cow, collections::HashSet, sync::Arc};
+
+use parking_lot::RwLock;
+use serde::{ser::SerializeStruct, Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::{clean_function_name, fetch_add_scope_id, short_file_name, ScopeId};
+
+#[derive(Default, Clone)]
+struct Inner {
+    // Store a both-way map, memory wise this can be a bit redundant but allows for faster access of information by external libs.
+    pub(crate) scope_id_to_details: std::collections::HashMap<ScopeId, ScopeDetails>,
+    pub(crate) string_to_scope_id: std::collections::HashMap<String, ScopeId>,
+}
+
+/// Provides fast read access to scope details.
+/// This collection can be cloned safely.
+#[derive(Default, Clone)]
+pub struct ScopeCollection(Arc<RwLock<Inner>>);
```

Comment:
Removing this will result in having to always have to lock the `GlobalProfiler` code even if the UI/userspace just wants to do a quick lookup of scope details. I thought it could be nice to have a read lock here as 99% of the usecases will involve read-only access.